themes/wds_headless/inc/options-page.php

```diff
@@ -0,0 +1,280 @@
+<?php
+/**
+ * Custom settings page for saving options for the theme.
+ *
+ * @author  WebDevStudios
+ * @package wds-headless-theme
+ * @since   1.0
+ */
+
+/*
+ * Utility Functions
+ * These functions are used to help make working with the options page easier.
+ */
+
+/**
+ * Determines if the constants are defined and, if so, disabled the option input fields
+ * except for the default 404 page.
+ *
+ * @author WebDevStudios
+ * @since  1.0.0
+ */
+function wds_headless_is_disabled_field() {
+	if ( _wds_headless_has_defined_all_constants() ) {
+		esc_attr_e( 'disabled="true"', 'wds' );
+	}
+}
+
+/**
+ * Read the incoming constant definition key and, if defined, updates the associated option.
+ *
+ * @access private
+ *
+ * @param string $wds_headless_constant The PHP constant to check before saving data from the form.
+ * @param string $option_name           The name of the option to serialize the value of the constant.
+ *
+ * @author WebDevStudios
+ * @since  1.0.0
+ */
+function _wds_headless_check_wp_config( string $wds_headless_constant, string $option_name ) {
+	$wds_headless_constant = strtoupper( $wds_headless_constant );
+	$constants             = _wds_headless_get_defined_constants();
+
+	return ( defined( $wds_headless_constant ) && isset( $constants['user'] ) );
+}
+
+/**
+ * Retrieves an associative array of constants defined in PHP.
+ *
+ * @access private
+ *
+ * @return array The associative array of PHP constants.
+ *
+ * @author WebDevStudios
+ * @since  1.0.0
+ */
+function _wds_headless_get_defined_constants() : array {
+	return \get_defined_constants( true );
+}
+
+/**
+ * Verifies that all of the WDS Headless constants are defined. If all of them are defined,
+ * this function will return true; otherwise, it will return false ignoring any that are
+ * defined.
+ *
+ * @access private
+ *
+ * @return bool True if all options are set in the wp-config.php file.
+ *
+ * @author WebDevStudios
+ * @since  1.0.0
+ */
+function _wds_headless_has_defined_all_constants() {
+	return (
+		_wds_headless_check_wp_config( 'HEADLESS_FRONTEND_URL', 'wds-headless-frontend-url' ) &&
+		_wds_headless_check_wp_config( 'WORDPRESS_PREVIEW_SECRET', 'wds-headless-preview-secret' ) &&
+		_wds_headless_check_wp_config( 'GRAPHQL_JWT_AUTH_SECRET_KEY', 'wds-headless-jwt-auth-key' )
+	);
```

Comment:
I'm curious as to the reasoning for making it all-or-nothing - is it just simplicity?

Reply:
Yes; @gregrickaby has mentioned trying to make this easy to get started for blue-collar WordPress developer or implementor. Since `wp-config.php` is something I assume you're either comfortable editing or not, then it made sense to either use it or don't.

In the future, though, I'm not opposed to having them individual. I'd want to be thoughtful as to which we'd consider (1 or 2 or all?), of course.